src/Access/Authentication.cpp

```diff
@@ -16,6 +25,59 @@ namespace ErrorCodes
     extern const int NOT_IMPLEMENTED;
 }
 
+// TODO: move to smth like utils.cpp file
+namespace
+{
+    std::vector<uint8_t> Base32Decoder(const std::string& input) {
```

Comment:
Could we reuse this decoder? https://github.com/ClickHouse/ClickHouse/blob/b4fa81f4a7648650902754f0f9bb9daf73e76928/src/Functions/GeoHash.cpp#L179